packages/collection/src/collection.ts

```diff
@@ -662,6 +662,21 @@ export class Collection<K, V> extends Map<K, V> {
 		return coll;
 	}
 
+	/**
+	 * The subtract method returns a new structure containing items where the keys and values of the original structure are not present in the other.
+	 *
+	 * @param other - The other Collection to filter against
+	 */
+	public subtract<T>(other: ReadonlyCollection<K, T>): Collection<K, V> {
+		const coll = new this.constructor[Symbol.species]<K, V>();
+		for (const [k, v] of this) {
+			if (!other.has(k) && !Object.is(v, other.get(k))) {
```

Comment:
Shouldn't this be an `||` instead? Or alternatively call .get once and check if its undefined or else if it equals to v via object.is?

Reply:
https://github.com/discordjs/discord.js/pull/8393#discussion_r933528705